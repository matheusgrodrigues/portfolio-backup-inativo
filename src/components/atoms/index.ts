import { AAvatar } from "./a-avatar/a-avatar";

export { AAvatar };
