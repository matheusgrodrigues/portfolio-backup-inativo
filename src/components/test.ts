import styled from "styled-components";

export const ButtonStyled = styled.button`
    background: yellow;
`;