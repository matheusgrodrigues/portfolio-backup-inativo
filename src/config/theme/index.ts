import { colors, font, spacing, bp } from './ref';

export const themeLight = {
    name: 'light',
    ref: { colors, font, spacing, bp },
};

export const darkTheme = {
    name: 'dark',
    ref: { colors, font, spacing, bp },
};
