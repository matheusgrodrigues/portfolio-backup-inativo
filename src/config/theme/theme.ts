const fontFamily = {
    fontFamily_inter: 'Inter',
};

export type FontSizeName = 'fontSize_textSm' | 'fontSize_textMd' | 'fontSize_textXl' | 'fontSize_displayLg';

const fontSize = {
    fontSize_textSm: '14px',
    fontSize_textMd: '16px',
    fontSize_textXl: '20px',
    fontSize_displayLg: '48px',
};

export type FontWeightName = 'fontWeight_semibold' | 'fontWeight_regular' | 'fontWeight_medium';

const fontWeight = {
    fontWeight_regular: '400',
    fontWeight_medium: '500',
    fontWeight_semibold: '600',
};

export type ColorName =
    | 'color_gradient_primary600'
    | 'color_primary500'
    | 'color_primary600'
    | 'color_gray500'
    | 'color_gray900'
    | 'color_gray50'
    | 'color_white'
    | 'color_black'
    | 'color_transparent';

const colors = {
    color_gradient_primary600: `linear-gradient(90deg, #7f56d9 0%, #9e77ed 100%);`,
    color_primary500: '#9e77ed',
    color_primary600: '#7f56d9',
    color_gray500: '#667085',
    color_gray900: '#101828',
    color_gray50: '#f9fafb',
    color_white: '#ffffff',
    color_black: '#000000',
    color_transparent: 'transparent',
};

export type SpacingName =
    | 'spacing_4'
    | 'spacing_8'
    | 'spacing_12'
    | 'spacing_16'
    | 'spacing_20'
    | 'spacing_24'
    | 'spacing_32'
    | 'spacing_40'
    | 'spacing_48'
    | 'spacing_64'
    | 'spacing_80'
    | 'spacing_96'
    | 'spacing_128'
    | 'spacing_160'
    | 'spacing_192'
    | 'spacing_224'
    | 'spacing_256';

const spacing = {
    spacing_4: '4px',
    spacing_8: '8px',
    spacing_12: '12px',
    spacing_16: '16px',
    spacing_20: '20px',
    spacing_24: '24px',
    spacing_32: '32px',
    spacing_40: '40px',
    spacing_48: '48px',
    spacing_64: '64px',
    spacing_80: '80px',
    spacing_96: '96px',
    spacing_128: '128px',
    spacing_160: '160px',
    spacing_192: '192px',
    spacing_224: '224px',
    spacing_256: '256px',
};

type BreakpointName = 'breakpoint_sm' | 'breakpoint_md' | 'breakpoint_lg' | 'breakpoint_xl';

const breakpoint = {
    breakpoint_sm: '640px',
    breakpoint_md: '768px',
    breakpoint_lg: '1024px',
    breakpoint_xl: '1280px',
};

export const bpHelper = (bp: BreakpointName, content: string) =>
    `@media screen and (min-width: ${breakpoint[bp]}) {
    ${content}
  }`;

export type ThemeName = 'light' | 'dark';

type Theme = {
    name: ThemeName;
    ref: {
        breakpoint: typeof breakpoint;
        fontFamily: typeof fontFamily;
        fontWeight: typeof fontWeight;
        fontSize: typeof fontSize;
        spacing: typeof spacing;
        colors: typeof colors;
    };
};

declare module 'styled-components' {
    export interface DefaultTheme extends Theme {}
}

export const themeLight = {
    name: 'light',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};

export const darkTheme = {
    name: 'dark',
    ref: { breakpoint, fontFamily, fontWeight, fontSize, spacing, colors },
};
