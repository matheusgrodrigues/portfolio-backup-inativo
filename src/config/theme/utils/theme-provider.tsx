'use client'

import { ThemeProvider as StyledThemeProvider } from 'styled-components'

export const ThemeProvider = StyledThemeProvider
