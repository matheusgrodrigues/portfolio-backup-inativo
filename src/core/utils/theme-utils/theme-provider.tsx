'use client';

import { ThemeProvider as StyledThemeProvider } from 'styled-components';

const ThemeProvider = StyledThemeProvider;

export default ThemeProvider;
