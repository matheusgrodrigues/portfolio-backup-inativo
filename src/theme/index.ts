import { por_ref_colors, por_ref_colorsDark } from "./ref";

const theme = {
  ref: { por_ref_colors },
  sys: {},
  comp: {},
};

export const darkTheme = {
  ...theme,
  ref: { ...theme.ref, por_ref_colors: { ...por_ref_colorsDark } },
};
