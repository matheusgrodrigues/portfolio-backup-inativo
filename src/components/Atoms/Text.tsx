import React from 'react'

import { ReactNode } from 'react'

import styled from 'styled-components'

import { themeLight } from '@/src/config/theme'
import { ThemeProvider } from '@/src/config/theme/helpers/theme-provider'

type TextFontWeigth = 'regular' | 'semibold'
type TextVariant = 'sm' | 'md' | 'lg' | 'xl'
type TextColor = 'white' | 'black' | 'gradient' | 'gray50' | 'gray500' | 'gray900'

interface TextProps {
    fontWeight: TextFontWeigth
    children: ReactNode
    $variant: TextVariant
    color: TextColor
}

/* eslint-disable no-unreachable */
const TextStyled = styled('p').attrs<TextProps>(() => ({}))`
    ${(props) => {
        const { fw_regular, fw_semibold } = props.theme.ref.font

        switch (props.fontWeight) {
            case 'regular':
                return `font-weight: ${fw_regular}`
                break
            case 'semibold':
                return `font-weight: ${fw_semibold}`
                break
            default:
                return `font-weight: initial`
                break
        }
    }}

    ${(props) => {
        const { fs_textSm, fs_textMd, fs_displayLg, fs_textXl } = props.theme.ref.font

        switch (props.$variant) {
            case 'lg':
                return `font-size: ${fs_displayLg};
        letter-spacing: -0.96px;`
                break
            case 'xl':
                return `font-size: ${fs_textXl}`
                break
            case 'md':
                return `font-size: ${fs_textMd}`
                break
            case 'sm':
                return `font-size: ${fs_textSm}`
                break
        }
    }};

    ${(props) => {
        const { color_gray50, color_primary500, color_primary600, color_gray900 } = props.theme.ref.colors

        switch (props.color) {
            case 'gradient':
                return `color: ${color_primary600}`
                break
            case 'gray50':
                return `color: ${color_gray50}`
            case 'gray500':
                return `color: ${color_primary500}`
                break
            case 'gray900':
                return `color: ${color_gray900}`
                break
        }
    }};
`

export const Text = ({ $variant, fontWeight, color, children }: TextProps) => {
    return (
        <ThemeProvider theme={themeLight}>
            <TextStyled data-testid="a-text" fontWeight={fontWeight} $variant={$variant} color={color}>
                {children}
            </TextStyled>
        </ThemeProvider>
    )
}
