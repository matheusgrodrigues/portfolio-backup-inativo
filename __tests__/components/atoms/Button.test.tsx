import renderer from 'react-test-renderer'

import { ThemeProvider } from '@/src/config/theme/helpers/theme-provider'
import { themeLight } from '@/src/config/theme'

import { Button } from '@/src/components/Atoms'

describe('Deve renderizar o Button corretamente', () => {
    it('Deve preservar a estrutura visual do componente', () => {
        const three = renderer
            .create(
                <ThemeProvider theme={themeLight}>
                    <Button
                        $background="color_gradient_primary600"
                        $fontWeight="fw_semibold"
                        $fontSize="fs_textMd"
                        $variant="lg"
                        $color="color_gray500"
                    >
                        AButton
                    </Button>
                </ThemeProvider>
            )
            .toJSON()

        expect(three).toMatchSnapshot()
    })
})
