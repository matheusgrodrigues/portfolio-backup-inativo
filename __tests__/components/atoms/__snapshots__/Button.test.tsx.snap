// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar o Button corretamente Deve preservar a estrutura visual do componente 1`] = `
<button
  className="sc-guDLey jfyRfX"
>
  AButton
</button>
`;
